packages/block-editor/src/components/tree-grid/README.md

```diff
@@ -0,0 +1,105 @@
+
+A tree grid allows the user to navigate using arrow keys. Up/down to navigate vertically across rows, and left/right to navigate horizontally between focusables in a row.
+
+For more information on a tree grid, see the following links:
+
+- https://www.w3.org/TR/wai-aria-practices/examples/treegrid/treegrid-1.html
+
+## Usage
+
+Wrap the tree with `<TreeGrid />`:
+
+```jsx
+function TreeMenu() {
+	return (
+		<TreeGrid>
+			<TreeGridRow level={ 1 } positionInSet={ 1 } setSize={ 2 }>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onSelect } { ...props }>Select</Button>
+				</TreeGridCell>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onMoveUp } { ...props }>Move Up</Button>
+				</TreeGridCell>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onMoveDown } { ...props }>Move Down</Button>
+				</TreeGridCell>
+			</TreeGridRow>
+			<TreeGridRow level={ 1 } positionInSet={ 2 } setSize={ 2 }>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onSelect } { ...props }>Select</Button>
+				</TreeGridCell>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onMoveUp } { ...props }>Move Up</Button>
+				</TreeGridCell>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onMoveDown } { ...props }>Move Down</Button>
+				</TreeGridCell>
+			</TreeGridRow>
+			<TreeGridRow level={ 2 } positionInSet={ 1 } setSize={ 1 }>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onSelect } { ...props }>Select</Button>
+				</TreeGridCell>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onMoveUp } { ...props }>Move Up</Button>
+				</TreeGridCell>
+				<TreeGridCell>
+					( props ) => <Button onClick={ onMoveDown } { ...props }>Move Down</Button>
+				</TreeGridCell>
+			</TreeGridRow>
+		</TreeGrid>
+	);
+}
+```
+
+# TreeGrid
+
+## Properties
+
+`TreeGrid` accepts no specific props. Any props specified will be passed to the `table` element rendered by `TreeGrid`.
+
+`TreeGrid` should always have children.
+
+# TreeGridRow
+
+## Properties
+
+Additional props other than those specified below will be passed to the `tr` element rendered by `TreeGridRow`, so for example, it is possible to also set a `className` on a row.
+
+### `level: number`
+
+An integer value designating the level in the hierarchical tree structure. Counting starts at 1. A value of `1` indicates the root level of the structure.
+
+### `positionInSet: number`
+
+An integer value that represents the position in the set. A set is the count of elements at a specific level. Counting starts at 1.
+
+### `setSize: number`
+
+An integer value that represents the total number of items in the set ... that is the total number of items at this specific level of the hierarchy.
+
+### `isExpanded: boolean`
+
+An optional value that designates whether a row is expanded or collapsed. Currently this value only sets the correct aria-expanded property on a row, it has no other built-in behaviour.
```

Comment:
Maybe I should just use the `aria-` equivalents of these props rather than aliasing them?

Reply:
Having those custom props may be useful for supporting React Native as well in the future.